js/src/components/app-button/index.js

```diff
@@ -49,9 +49,14 @@ const AppButton = ( props ) => {
 		onClick( ...args );
 	};
 
+	const classes = [ 'app-button', className ];
+	if ( rest.iconPosition === 'right' ) {
+		classes.push( 'gla-app-button--fix-icon-position' );
+	}
+
 	return (
 		<Button
-			className={ classnames( 'app-button', className ) }
+			className={ classnames( ...classes ) }
```

Comment:
💅  `classnames` [API](https://www.npmjs.com/package/classnames) is pretty flexible and lenient, you can have it all in one line like the following: (note that I'm putting the two "app-button" class together.)

```suggestion
			className={ classnames(
				'app-button',
				{
					'app-button__icon-position-right':
						rest.iconPosition === 'right',
				},
				className
			) }
```

Reply:
Thanks! I knew it but I was trying to keep the code easier to read. From my perspective, glue too much logic within JSX does not look better to read.

Reply:
My perspective is that with this suggested change, we won't need to have [Line 52-55](https://github.com/woocommerce/google-listings-and-ads/pull/918/files#diff-ec24036234748771009e50c0fe89c4d4ba61fb4af1484a5162b342e2f1a46640R52-R55) (declaring `classes` array variable and conditionally pushing element into it) which IMO is taking a lil bit more mental energy to read and maintain.

Anyway, this is personal preference 💅 , not a deal breaker, OK to proceed with what we have. 👌